synchronizer/synchronizer.go

```diff
@@ -110,41 +121,26 @@ func (s *ClientSynchronizer) Sync() error {
 					continue
 				}
 			}
-			if waitDuration != s.cfg.SyncInterval.Duration {
-				latestsequencedBatchNumber, err := s.etherMan.GetLatestBatchNumber()
-				if err != nil {
-					log.Warn("error getting latest sequenced batch in the rollup. Error: ", err)
-					continue
-				}
-				// Check latest Synced Batch
-				dbTx, err := s.state.BeginStateTransaction(s.ctx)
-				if err != nil {
-					log.Fatalf("error creating db transaction to get latestSyncedBatch. error: %s", err.Error())
-				}
-				latestSyncedBatch, err := s.state.GetLastBatchNumber(s.ctx, dbTx)
-				errC := dbTx.Commit(s.ctx)
-				if errC != nil {
-					log.Errorf("error committing dbTx, err: %s", errC.Error())
-					rollbackErr := dbTx.Rollback(s.ctx)
-					if rollbackErr != nil {
-						log.Fatalf("error rolling back state. RollbackErr: %s, err: %s",
-							rollbackErr.Error(), errC.Error())
-					}
-					log.Fatalf("error committing dbTx, err: %s", errC.Error())
-				}
+			latestSequencedBatchNumber, err := s.etherMan.GetLatestBatchNumber()
+			if err != nil {
+				log.Warn("error getting latest sequenced batch in the rollup. Error: ", err)
+				continue
+			}
+			latestSyncedBatch, err := s.state.GetLastBatchNumber(s.ctx, nil)
+			if err != nil {
+				log.Warn("error getting latest batch synced. Error: ", err)
+				continue
+			}
+			if latestSyncedBatch >= latestSequencedBatchNumber {
```

Comment:
I don't thing doing these operations all the time even if they are not necessary, is worth it. I suggest to keep the if  `			if waitDuration != s.cfg.SyncInterval.Duration {`

Reply:
The behavior of the synchronizer now is different from before because the old implementation was considering data coming only from L1, but now we have data coming from L1(virtual and verified) and L2(trusted).

The method `s.syncBlocks` is the one that syncs the L1(virtual and verified) data, but this method doesn't know if all the L1 data was synced, so the sync process after syncing part of the L1 needs to check the last batch in the L2 state and, the last batch in the rollup to decide if its time to start syncing data from L2(trusted).